Data Analysis Step 2 Abnormal Behaviour Identification/Data Analysis Step 2 Abnormal Behaviour Identification on Laballed Data.ipynb

```diff
@@ -1809,6 +1809,139 @@
+    "\n",
+    "Accel_4['Amag'] = np.sqrt((Accel_4['X'])**2 + (Accel_4['Y'])**2 + (Accel_4['Z'])**2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "41e8484f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Preview the data. \n",
+    "\n",
+    "Accel_4.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1896a1f5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate normalised fAmag & Amag for Acceleration 4. \n",
+    "\n",
+    "Accel_4['fAmag_norm'] = ((Accel_4['fAmag'] - Accel_4['fAmag'].min())/ \n",
+    "                                     (Accel_4['fAmag'].max() - Accel_4['fAmag'].min()))\n",
+    "\n",
+    "Accel_4['Amag_norm'] = ((Accel_4['Amag'] - Accel_4['Amag'].min())/ \n",
+    "                                     (Accel_4['Amag'].max() - Accel_4['Amag'].min()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f490985f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Preview the data.\n",
+    "\n",
+    "Accel_4.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8bc30899",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Apply the different thresholds to Acceleration 4 data. \n",
+    "\n",
+    "# Q3 (75th percentile) threshold\n",
+    "Accel_4['Q3 threshold'] = Accel_4['Amag_norm'].apply(Q3_thresholds)\n",
+    "\n",
+    "# Standard deviation threshold\n",
+    "Accel_4['Std threshold'] = Accel_4['Amag_norm'].apply(std_threshold)\n",
+    "\n",
+    "# High Outlier threshold\n",
+    "Accel_4['HO threshold'] = Accel_4['Amag_norm'].apply(HO_threshold)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b96cfa08",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Preview the data.\n",
+    "\n",
+    "Accel_4.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "351fd171",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualise the simple classification of normalised fAmag data.\n",
+    "\n",
+    "hue_order = ['Normal data', 'Abnormal data']\n",
+    "\n",
+    "\n",
+    "fig, axes = plt.subplots(3,1, figsize=(12,10), sharex=True, sharey=True)\n",
+    "\n",
+    "sns.scatterplot(data=Accel_4, ax=axes[0], x = 'Time', y = 'Amag_norm', hue = 'Q3 threshold', hue_order= hue_order).set(title = 'Acceleration 4 - Q3 Threshold')\n",
+    "axes[0].set_xlabel('')\n",
+    "\n",
+    "sns.scatterplot(data=Accel_4, ax=axes[1], x = 'Time', y = 'Amag_norm', hue = 'Std threshold', hue_order= hue_order).set(title = 'Acceleration 4 - Std threshold')\n",
+    "axes[1].set_xlabel('')\n",
+    "\n",
+    "sns.scatterplot(data=Accel_4, ax=axes[2], x = 'Time', y = 'Amag_norm', hue = 'HO threshold', hue_order= hue_order).set(title = 'Acceleration 4 - HO threshold')\n"
+   ]
+  },
```

Comment:
If you dont mind, i can inlude your part accel 4 in the file but then you would not be seen as contributor to repisatory.